src/rosdep2/platforms/opensuse.py

```diff
@@ -44,14 +45,15 @@ def register_installers(context):
 
 def register_platforms(context):
     context.add_os_installer_key(OS_OPENSUSE, SOURCE_INSTALLER)
+    context.add_os_installer_key(OS_OPENSUSE, PIP_INSTALLER)
     context.add_os_installer_key(OS_OPENSUSE, ZYPPER_INSTALLER)
     context.set_default_os_installer_key(OS_OPENSUSE, lambda self: ZYPPER_INSTALLER)
 
 
 def rpm_detect(packages):
     installed = []
     for p in packages:
-        if not subprocess.call(['rpm', '-q', p]):
+        if not subprocess.call(['rpm', '-q', '--whatprovides', p]):
```

Comment:
It looks like the rpm facilities for the redhat platform are a bit more sophisticated. Is it worth centralizing and sharing some of this logic or is there too much divergence between them? Also asking @cottsay as my local RedHat expert and probable contributor for the redhat RPM handling.

Reply:
This is a similar approach to the one taken in the RedHat installer:
https://github.com/ros-infrastructure/rosdep/blob/cfa21c89d440e3fcb6c41b1f4a5f4947a5649b41/src/rosdep2/platforms/redhat.py#L110-L118

Invoking the `rpm` executable repetitively on a large workspace was pretty slow, so a single command is invoked for all of the packages and the output is parsed to ensure availability since the return value isn't useful anymore.

This change, however, shouldn't make the openSUSE installer any less performant than it currently is, and uses an already proven approach.

Reply:
> Is it worth centralizing and sharing some of this logic...?

Maybe. The RedHat installer tries to use the RPM python module if it is available, so I think we'd need to do some testing to ensure it works right, but I expect that the RPM solution in `redhat.py` would work here as-is.

I don't think that work should block this PR though.